### PR TITLE
feat(spanner): tests and samples for drop-database protection

### PR DESCRIPTION
**Pull request by devbww**
Exercise the new `DatabaseAdminClient::UpdateDatabase()` interface to set the `Database.enable_drop_protection` field.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11637)

### CODE REVIEW

File: google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc [hunk 1]
```diff
@@ -581,6 +581,69 @@ TEST_F(DatabaseAdminClientTest, DatabasePostgreSQLBasics) {
...
+  EXPECT_FALSE(database->enable_drop_protection());
+  EXPECT_FALSE(database->reconciling());
+
+  google::protobuf::FieldMask update_mask;
+  update_mask.add_paths("enable_drop_protection");
+
+  database->set_enable_drop_protection(true);
+  auto updated = client_.UpdateDatabase(*database, update_mask).get();
+  if (emulator_) {
+    EXPECT_THAT(updated, StatusIs(StatusCode::kUnimplemented));
```
Comment (coryan) [inline google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc:603]: Consider just skipping the test in this case. We already have tests for `CreateDatabase()` and `DropDatabase()`.

File: google/cloud/spanner/samples/samples.cc [hunk 2]
```diff
@@ -1537,6 +1537,35 @@ void ListDatabaseOperationsCommand(std::vector<std::string> argv) {
...
+
+void UpdateDatabaseCommand(std::vector<std::string> argv) {
+  if (argv.size() != 4) {
+    throw std::runtime_error(
+        "update-database <project-id> <instance-id> <database-id>"
+        " <drop-protection>");
+  }
+  google::cloud::spanner_admin::DatabaseAdminClient client(
+      google::cloud::spanner_admin::MakeDatabaseAdminConnection());
+  bool drop_protection = (argv[3][0] == 'T');
```
Comment (coryan) [inline google/cloud/spanner/samples/samples.cc:1565]: assuming `! argv[3].empty()` ?

**Review by coryan — APPROVED** [review]

File: google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc (hunk 1 above)
Comment (devbww) [reply google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc:603]: The idea there is to get some active notification of when the behavior of the emulator changes.

We do have some instances of `if (emulator) GTEST_SKIP()`, but they are either old, or where the above tactic is too complicated.

File: google/cloud/spanner/samples/samples.cc (hunk 2 above)
Comment (devbww) [reply google/cloud/spanner/samples/samples.cc:1565]: `s[i]` is actually [and weirdly] defined to return (a reference to) a `char()` value when `i == s.size()`.

File: google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc [hunk 3]
```diff
@@ -581,6 +581,69 @@ TEST_F(DatabaseAdminClientTest, DatabasePostgreSQLBasics) {
   EXPECT_STATUS_OK(drop_status);
 }
 
+/// @test Verify dropping a database fails with drop protection enabled.
+TEST_F(DatabaseAdminClientTest, DropDatabaseProtection) {
```
[the remark this answers is not in the document]
Comment (devbww) [reply google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc:585]: I'm (almost) totally with you.  It does exercise our generated code a little, but the overarching reason for it is to cover a "thou shalt have" clause in the feature-specification doc that isn't worth contesting.  :-|

File: google/cloud/spanner/samples/samples.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/spanner/samples/samples.cc:1565]: TIL

File: google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc:603]: Okay.

File: google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc (hunk 3 above)
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc:585]: I think it is worth contesting, but not blocking this PR for it.

[the remark this answers is not in the document]
Comment (devbww) [reply google/cloud/spanner/admin/integration_tests/database_admin_integration_test.cc:585]: Ack.

**Pull request merged**